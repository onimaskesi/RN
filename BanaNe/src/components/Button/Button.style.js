import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

const base = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
    borderRadius: 10,
    borderWidth: 1,
  },
  icon: {
    margin: 5,
  },
  title: {
    fontSize: 18,
    margin: 5,
  },
});

const primary = StyleSheet.create({
  ...base,
  container: {
    ...base.container,
    backgroundColor: colors.green,
    borderColor: 'white',
  },
  title: {
    ...base.title,
    color: 'white',
  },
});

const secondary = StyleSheet.create({
  ...base,
  container: {
    ...base.container,
    backgroundColor: colors.lightGreen,
    borderColor: colors.green,
  },
  title: {
    ...base.title,
    color: colors.green,
  },
});

const secondary_active = StyleSheet.create({
  ...secondary,
  container: {
    ...secondary.container,
    borderColor: colors.darkGreen,
  },
  title: {
    ...secondary.title,
    color: colors.darkGreen,
  },
});

const primary_active = StyleSheet.create({
  ...primary,
  container: {
    ...primary.container,
    backgroundColor: colors.darkGreen,
  },
});

export default {primary, secondary, primary_active, secondary_active};
