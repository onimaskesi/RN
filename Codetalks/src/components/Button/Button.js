import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './Button.style';

export default ({onPress, title, iconName, theme = 'primary', disabled}) => {
  const style = disabled ? styles[theme] : styles[`${theme}_active`];

  return (
    <TouchableOpacity
      style={style.container}
      onPress={onPress}
      disabled={disabled}>
      {iconName && (
        <Icon style={style.icon} name={iconName} size={20} color="white" />
      )}
      <Text style={style.title}>{title}</Text>
    </TouchableOpacity>
  );
};
