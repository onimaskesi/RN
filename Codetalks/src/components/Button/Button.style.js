import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

const base = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
    borderRadius: 10,
    borderWidth: 1,
  },
  icon: {
    margin: 5,
  },
  title: {
    fontSize: 18,
    margin: 5,
  },
});

const primary = StyleSheet.create({
  ...base,
  container: {
    ...base.container,
    backgroundColor: colors.orange,
    borderColor: 'white',
  },
  title: {
    ...base.title,
    color: 'white',
  },
});

const secondary = StyleSheet.create({
  ...base,
  container: {
    ...base.container,
    backgroundColor: colors.lightOrange,
    borderColor: colors.orange,
  },
  title: {
    ...base.title,
    color: colors.orange,
  },
});

const secondary_active = StyleSheet.create({
  ...secondary,
  container: {
    ...secondary.container,
    borderColor: colors.brown,
  },
  title: {
    ...secondary.title,
    color: colors.brown,
  },
});

const primary_active = StyleSheet.create({
  ...primary,
  container: {
    ...primary.container,
    backgroundColor: colors.brown,
  },
});

export default {primary, secondary, primary_active, secondary_active};
