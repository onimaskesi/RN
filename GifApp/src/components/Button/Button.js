import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import styles from './ButtonStyle';

export default ({onClick, title}) => {
  return (
    <TouchableOpacity onPress={onClick} style={styles.container}>
      <Text style={styles.title}>{title}</Text>
    </TouchableOpacity>
  );
};
