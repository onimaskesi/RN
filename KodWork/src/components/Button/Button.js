import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './Button.style';

export default ({onPress, title, iconName}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.inner_container}>
        <Icon style={styles.icon} name={iconName} size={20} color="white" />
        <Text style={styles.title}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};
