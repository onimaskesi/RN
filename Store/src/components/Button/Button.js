import React from 'react';
import {TouchableOpacity, Text, ActivityIndicator} from 'react-native';
import styles from './Button.style';

export default ({title, onPress, isLoading}) => (
  <TouchableOpacity
    style={styles.container}
    onPress={onPress}
    disabled={isLoading}>
    {isLoading ? (
      <ActivityIndicator size="large" color="white" />
    ) : (
      <Text style={styles.title}>{title}</Text>
    )}
  </TouchableOpacity>
);
