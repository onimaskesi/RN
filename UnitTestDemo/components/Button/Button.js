import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';
import styles from './Button.style';

export default function ({title, onClick}) {
  return (
    <TouchableOpacity
      testID="button-touchable"
      style={styles.container}
      onPress={onClick}>
      <Text testID="button-title" style={styles.title}>
        {title}
      </Text>
    </TouchableOpacity>
  );
}
