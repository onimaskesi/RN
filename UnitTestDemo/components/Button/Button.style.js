import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    padding: 10,
    margin: 10,
    backgroundColor: 'blue',
    borderRadius: 10,
  },
  title: {
    color: 'white',
  },
});
