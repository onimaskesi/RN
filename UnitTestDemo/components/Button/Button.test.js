import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from './Button';
import styles from './Button.style';

test('should match with snapshot', () => {
  const comp = render(<Button />);
  expect(comp).toMatchSnapshot();
});

test('should render title correctly', () => {
  const titleText = 'test';
  const comp = render(<Button title={titleText} />);

  const buttonText = comp.getByTestId('button-title').children[0];

  expect(buttonText).toBe(titleText);
});

test('should triggered onPress', () => {
  const mockFunction = jest.fn();
  const comp = render(<Button onClick={mockFunction} />);

  const buttonTouchable = comp.getByTestId('button-touchable');
  fireEvent(buttonTouchable, 'press');

  expect(mockFunction).toBeCalled();
});

test('should render default theme style as primary style', () => {
  const comp = render(<Button />);

  const buttonStyle = comp.getByTestId('button-touchable').props.style;
  expect(buttonStyle).toMatchObject(styles.primary.container);
});

test('should render given theme style', () => {
  const selectedTheme = 'secondary';
  const comp = render(<Button theme={selectedTheme} />);

  const buttonStyle = comp.getByTestId('button-touchable').props.style;
  expect(buttonStyle).toMatchObject(styles[selectedTheme].container);
});
