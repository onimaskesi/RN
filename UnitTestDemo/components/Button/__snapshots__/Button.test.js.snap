// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "borderRadius": 10,
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
    testID="button-title"
  />
</View>
`;
